### PR TITLE
[Merged by Bors] - Add documentation to the WindowDescriptor struct.

### LINKED ISSUE #4753: Add docs to WindowDescriptor
## How can Bevy's documentation be improved?

`WindowDescriptor` has no docs. It is a very important struct for beginners, because it's used for common tasks like setting the screen size and changing the `PresentMode`.

Each of the fields should have their own documentation.

### PR DESCRIPTION
**Pull request by object71**
# Objective

Resolves #4753

## Solution

Using rust doc I added documentation to the struct. Decided to not provide an example in the doc comment but instead refer to the example file that shows the usage.

### CODE REVIEW

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:606]:
```suggestion
///
/// This should be set up before adding the [`WindowPlugin`]. Most of these settings can also later
```

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:605]:
```suggestion
/// Describes the information needed for a window creation.
```
This information is redundant to the info in the field.

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:609]:
```suggestion
/// See [`examples/window/window_settings.rs`](https://github.com/bevyengine/bevy/blob/main/examples/window/window_settings.rs) for usage.
```

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:615]: What does "logical width" mean more precisely? Is this pixels? pixels modified by the scale?

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:619]:
```suggestion
    /// The position on the screen that the window will be centered at.
```

File: crates/bevy_window/src/window.rs
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:624]: I'd prefer to just embed this directly, using `[Window](crate::window)` rather than splitting it across lines.

File: crates/bevy_window/src/window.rs
Comment (mockersf) [inline crates/bevy_window/src/window.rs:645]: This field is changed a little and modified in https://github.com/bevyengine/bevy/pull/4726

File: crates/bevy_window/src/window.rs
Comment (Nilirad) [reply crates/bevy_window/src/window.rs:609]: I think the best format for this link is reference-style, due to the length of the URL.

File: crates/bevy_window/src/window.rs
Comment (object71) [reply crates/bevy_window/src/window.rs:616]: None of the other supporting classes mentions what it is about but the ```Window``` class has a full description of the sizes - logical and physical. I just wanted to mention here that it is logical size and not bother with copying the whole section out of the ```Window``` class.

File: crates/bevy_window/src/window.rs
Comment (object71) [reply crates/bevy_window/src/window.rs:641]: I will check it out.

Comment (object71) [reply crates/bevy_window/src/window.rs:641]: Looking at the change there's nothing that doesn't reflect the current documentation that I am adding.

File: crates/bevy_window/src/window.rs
Comment (object71) [reply crates/bevy_window/src/window.rs:613]: Added one more line of description why it might vary.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@bevyengine/docs-team can I get a second review?

File: crates/bevy_window/src/window.rs
Comment (IceSentry) [inline crates/bevy_window/src/window.rs:619]: I think this should explain a bit what happens when you pass it None

This is what the winit docs says:
```rust
    /// If this is `None`, some platform-specific position
    /// will be chosen.
    ///
    /// The default is `None`.
    ///
    /// ## Platform-specific
    ///
    /// - **macOS**: The top left corner position of the window content, the window's "inner"
    /// position. The window title bar will be placed above it.
    /// The window will be positioned such that it fits on screen, maintaining
    /// set `inner_size` if any.
    /// If you need to precisely position the top left corner of the whole window you have to
    /// use [`Window::set_outer_position`] after creating the window.
    /// - **Windows**: The top left corner position of the window title bar, the window's "outer"
    /// position.
    /// There may be a small gap between this position and the window due to the specifics of the
    /// Window Manager.
    /// - **X11**: The top left corner of the window, the window's "outer" position.
    /// - **Others**: Ignored.
    ///
```

Maybe just mentioning that the default position is controlled by the platform when None is used

Comment (object71) [reply crates/bevy_window/src/window.rs:619]: Mentioned it!

**Comment by object71** [comment]
Also added a short comment for decorations in the last change as it isn't descriptive at first glance (or at least for me). So I searched it in winit and found that it is responsible for borders and bars.

**Comment by alice-i-cecile** [comment]
Awesome, thanks again! BTW, in the future, prefer many small commits over rebasing when contributing here. We squash on merge, and the separate commits make it easier for reviewers to quickly check what's changed.

**Comment by alice-i-cecile** [comment]
> [✖] http://blog.izs.me/post/30036893703/policy-on-trolling → Status: 0 Error: ETIMEDOUT
>     at Timeout.<anonymous> (/usr/local/lib/node_modules/markdown-link-check/node_modules/request/request.js:848:19)
>     at listOnTimeout (node:internal/timers:564:17)
>     at process.processTimers (node:internal/timers:507:7) {
>   code: 'ETIMEDOUT',
>   connect: true
> }
> [✖] https://www.contributor-covenant.org/ → Status: 0 Error: ETIMEDOUT
>     at Timeout.<anonymous> (/usr/local/lib/node_modules/markdown-link-check/node_modules/request/request.js:848:19)
>     at listOnTimeout (node:internal/timers:564:17)
>     at process.processTimers (node:internal/timers:507:7) {
>   code: 'ETIMEDOUT',
>   connect: true
> }

Ah, #4575 strikes again. Not your fault here.

**Review by IceSentry — APPROVED** [review]

**Comment by object71** [comment]
> y check what's changed.

Ok, thanks for the guideline :)

File: crates/bevy_window/src/window.rs
Comment (Nilirad) [inline crates/bevy_window/src/window.rs:632]:
```suggestion
    /// Controls when a frame is presented to the screen.
```

This wording is easier to understand.

**Review by Nilirad — CHANGES_REQUESTED** [review]
Great work with documenting all fields!

Once the wording suggestion below is applied, I will feel comfortable to approve this PR. 😄

**Review by Nilirad — APPROVED** [review]

File: crates/bevy_window/src/window.rs
Comment (IceSentry) [inline crates/bevy_window/src/window.rs:649]: So after some discussion on discord, we discovered that on X11 the default scale factor is a bit weird and doesn't always defaults to 1.0.

See the thread here https://discord.com/channels/691052431525675048/977977143004905584/978161092155031552

The fix was to force a scale factor override of 1.0. Just reading the winit docs makes it pretty clear that this is a complicated issue for X11 https://docs.rs/winit/latest/winit/dpi/index.html#how-is-the-scale-factor-calculated

I think it would be a good idea to at least mention here that if users are having unexpected scaling when using X11 they should try setting it to Some(1.0)

**Comment by object71** [comment]
I don't understand why it fails. Do I need to do something else?

**Comment by alice-i-cecile** [comment]
Not your fault; this was a flaky check that we've since removed.

**Pull request closed without merging**
